src/middlewared/middlewared/plugins/vmware.py

```diff
@@ -194,7 +193,16 @@ def get_datastores(self, data):
             datastores[esxi_host.name] = datastores_host
 
         connect.Disconnect(server_instance)
-        return datastores
+
+        # Datastore names are unique among different esxi host under a single vcenter server. As we only require
+        # datastore names and additional information is not needed for datastore choices, we refine the datastores
+        # dict and send back only the datastore names
+
+        return OrderedDict({
```

Comment:
Hmm as far as I know JSON does not guaranteed order in objects, why OrderedDict then?

Reply:
Uh oh, i missed that. Yes this will not work in that case. My initial intention was to try to keep the dictionary convention for choices, but in this case i think i'll use a list then. Arrays are preserved in json